### PR TITLE
fix: Close panels when object is nulled out

### LINKED ISSUE #167: deephaven.ui components do not close if you null out the component
**Description**

When you null out another type of component, it closes. However, deephaven.ui doesn't close.

**Steps to reproduce**
1. Execute the following snippet to create a table and a deephaven.ui component:
```
from deephaven import empty_table, ui

@ui.component
def counter():
    count, set_count = ui.use_state(0)
    return ui.action_button(
        f"You pressed me {count} times", on_press=lambda: set_count(count + 1)
    )


c = counter()
t = empty_table(100).update("x=i")
```
2. Execute the following snippet to null out the table and custom component:
```
c = None
t = None
```

**Expected results**
2. Both t and c should close

**Actual results**
2. Only t closes

### PR DESCRIPTION
**Pull request by mofojed**
- Needed to listen for the `PanelEvent.CLOSE` event
- Fixes #167

### CODE REVIEW

File: plugins/ui/src/js/src/DashboardPlugin.tsx [hunk 1]
```diff
@@ -64,6 +64,17 @@ export function DashboardPlugin({
...
+  const handlePanelClose = useCallback((panelId: string) => {
+    setWidgetMap(prevWidgetMap => {
+      if (!prevWidgetMap.has(panelId)) {
+        return prevWidgetMap;
+      }
+      const newWidgetMap = new Map<string, WidgetWrapper>(prevWidgetMap);
+      newWidgetMap.delete(panelId);
+      return newWidgetMap;
+    });
+  }, []);
```
Comment (mattrunyon) [inline plugins/ui/src/js/src/DashboardPlugin.tsx:76]: Should this close all panels in the widget map instead? I think this is really if the document is closed/deleted so it should close all panels/widgets in the document

**Review by mattrunyon — CHANGES_REQUESTED** [review]

File: plugins/ui/src/js/src/DashboardPlugin.tsx (hunk 1 above)
Comment (mofojed) [reply plugins/ui/src/js/src/DashboardPlugin.tsx:76]: Each open widget has one document which can have multiple panels. This is closing the top level widget which will unmount/close all panels opened by that widget.

Comment (mofojed) [reply plugins/ui/src/js/src/DashboardPlugin.tsx:76]: I tested with the following snippet opening multiple panels:
```
from deephaven import ui
from deephaven.plot import express as dx

_stocks = dx.data.stocks()

@ui.component
def stock_table_input(source, default_sym="", default_exchange=""):
    sym, set_sym = ui.use_state(default_sym)
    exchange, set_exchange = ui.use_state(default_exchange)

    t1 = source.select_distinct(["sym", "exchange"])
    t2 = source.where([f"sym=`{sym.upper()}`", f"exchange=`{exchange}`"])
    p = dx.line(t2, x="timestamp", y="price")

    def handle_row_double_press(row, data):
        set_sym(data["sym"]["value"])
        set_exchange(data["exchange"]["value"])

    return [
        ui.panel(
            # Add a callback for when user double clicks a row in the table
            ui.table(t1).on_row_double_press(handle_row_double_press),
            title="Stock Table Input",
        ),
        ui.panel(t2, title="Stock Filtered Table"),
        ui.panel(p, title="Stock Plot"),
    ]


sti = stock_table_input(_stocks, "CAT", "TPET")
```
And then `sti = None`. All panels cleared out.

Comment (mattrunyon) [reply plugins/ui/src/js/src/DashboardPlugin.tsx:76]: Oh ya for some reason I thought this was in the document handler

**Review by mattrunyon — APPROVED** [review]

**Pull request merged**